### ISSUE TITLE
Legion: non informative error in case of impossible compact sparse instance covering

### ISSUE THREAD

**Issue by mariodirenzo** [StanfordLegion/legion#1388]
If the call to `compute_covering` (https://gitlab.com/StanfordLegion/legion/-/blob/control_replication/runtime/legion/region_tree.inl#L372) fails during the creation of a compact sparse instance, the runtime ends up computing the potential memory footprint of the instance using a `NULL` layout. As a result, the default mapper issues errors like
```
[0 - 7f525eff6840]  169.168499 {5}{default_mapper}: Default mapper failed allocation of size 140473362549674 bytes for region  requirement 1 of task UpdateUsingDiffusionFluxZ (UID 7548) in memory 1e00000000000000 (SYSTEM_MEM) for processor 1d00000000000003 (OMP_PROC). This means the working set of your application is too big for the allotted capacity of the given memory under the default mapper's mapping scheme. You have three choices: ask Realm to allocate more memory, write a custom mapper to better manage working sets, or find a bigger machine.
```
whereby a non-sensical memory requirement is reported.
Having a more informative error would help the debugging of the layout constraints.

**Comment by lightsighter** [comment]
I think I will need a reproducer because Legion I can't see how Legion can return something other than `0` or the right size of the instance. We initialize the instance layout size here:

https://gitlab.com/StanfordLegion/legion/-/blob/control_replication/runtime/legion/region_tree.inl#L350

And that should propagate all the way back out to the footprint that the mapper sees if the covering computation fails. This probably also needs to be fixed, but that means we still don't understand how you get a crazy large size for the instance, and we need to understand that before fixing anything.

**Comment by mariodirenzo** [comment]
The issue has been fixed in 90085a37.
Thank you very much!
